pkg/beholder/client.go

```diff
@@ -0,0 +1,305 @@
+package beholder
+
+import (
+	"context"
+	"errors"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/otlp/otlplog/otlploggrpc"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetricgrpc"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	otellog "go.opentelemetry.io/otel/log"
+	otelmetric "go.opentelemetry.io/otel/metric"
+	sdklog "go.opentelemetry.io/otel/sdk/log"
+	sdkmetric "go.opentelemetry.io/otel/sdk/metric"
+	sdkresource "go.opentelemetry.io/otel/sdk/resource"
+	"go.opentelemetry.io/otel/sdk/trace"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	oteltrace "go.opentelemetry.io/otel/trace"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+type MessageEmitter interface {
+	// Sends message with bytes and attributes to OTel Collector
+	Emit(ctx context.Context, body []byte, attrKVs ...any) error
+}
+type Client interface {
+	Close() error
+}
+
+type messageEmitter struct {
+	messageLogger otellog.Logger
+}
+
+type OtelClient struct {
+	Config Config
+	// Logger
+	Logger otellog.Logger
+	// Tracer
+	Tracer oteltrace.Tracer
+	// Meter
+	Meter otelmetric.Meter
+	// Message Emitter
+	Emitter MessageEmitter
+
+	// Providers
+	LoggerProvider        otellog.LoggerProvider
+	TracerProvider        oteltrace.TracerProvider
+	MeterProvider         otelmetric.MeterProvider
+	MessageLoggerProvider otellog.LoggerProvider
+
+	// OnClose
+	OnClose func() error
+}
+
+// NewOtelClient creates a new Client with OTel exporter
+func NewOtelClient(ctx context.Context, cfg Config, errorHandler errorHandlerFunc) (OtelClient, error) {
```

Comment:
We should either export this error handler type:
```suggestion
func NewOtelClient(ctx context.Context, cfg Config, errorHandler ErrorHandlerFunc) (OtelClient, error) {
```
Or just inline the literal since it is so simple:
```suggestion
func NewOtelClient(ctx context.Context, cfg Config, errorHandler func(error)) (OtelClient, error) {
```

Reply:
Oh, actually this is only used to set the global otel error handler. I think that is a separate concern like the other global setters that we dropped. Let's remove it from here. We could doc that it is important to set, but there is no need to entangle it with otel client construction.

Reply:
Addressed https://github.com/smartcontractkit/chainlink-common/pull/694/commits/26c09d7b0c62e7fdf69559393440bb076ac84864